src/profile-logic/profile-data.js

```diff
@@ -576,41 +588,40 @@ function _compareNonInvertedCallNodesInSuffixOrder(
 // that frame, for example the frame's address or line.
 export function getMatchingAncestorStackForInvertedCallNode(
   needleStack: IndexIntoStackTable,
-  invertedTreeCallNode: IndexIntoCallNodeTable,
-  invertedTreeCallNodeSubtreeEnd: IndexIntoCallNodeTable,
+  suffixOrderIndexRangeStart: SuffixOrderIndex,
+  suffixOrderIndexRangeEnd: SuffixOrderIndex,
+  suffixOrderIndexes: Uint32Array,
   invertedTreeCallNodeDepth: number,
-  stackIndexToInvertedCallNodeIndex: Int32Array,
+  stackIndexToCallNodeIndex: Int32Array,
   stackTablePrefixCol: Array<IndexIntoStackTable | null>
```

Comment:
Good point, I agree these arguments could be consolidated a bit.

Reply:
Actually, I looked at this again and the function is called for every sample. I think I'd like to keep it the way it is so that as much duplicate work as possible is avoided.